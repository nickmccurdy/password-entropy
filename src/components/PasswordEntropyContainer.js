import React, { Component } from 'react'
import PasswordEntropy from './PasswordEntropy'

export default class PasswordEntropyContainer extends Component {
  state = {
    length: 12,
    options: {
      Custom: 0,
      Generic: {
        Letters: true,
        'Capital Letters': true,
        Numbers: true,
        Symbols: true,
      },
    },
    optionsKey: 'Generic',
  }

  handleCustom = ({ target: { value } }) =>
    this.setState(({ options }) => ({
      options: { ...options, Custom: parseInt(value, 10) },
    }))

  handleGeneric = ({ target: { checked, name } }) =>
    this.setState(({ options: { Custom, Generic } }) => ({
      options: { Custom, Generic: { ...Generic, [name]: checked } },
    }))

  handleLength = ({ target: { value } }) =>
    this.setState({ length: parseInt(value, 10) })

  handleOptionsKey = (optionsKey) => this.setState({ optionsKey })

  render() {
    const { length, options, optionsKey } = this.state

    return (
      <PasswordEntropy
        length={length}
        onCustom={this.handleCustom}
        onGeneric={this.handleGeneric}
        onLength={this.handleLength}
        onOptionsKey={this.handleOptionsKey}
        options={options}
        optionsKey={optionsKey}
      />
    )
  }
}
