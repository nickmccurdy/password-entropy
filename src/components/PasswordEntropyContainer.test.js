import { shallow } from 'enzyme'
import React from 'react'

import PasswordEntropyContainer from './PasswordEntropyContainer'

test('PasswordEntropyContainer', () => {
  const wrapper = shallow(<PasswordEntropyContainer />)
  expect(wrapper).toMatchSnapshot()
})

test('handleCustom', () => {
  const wrapper = shallow(<PasswordEntropyContainer />)
  wrapper.find('PasswordEntropy').simulate('custom', { target: { value: '1' } })
  expect(wrapper.state()).toMatchObject({ options: { Custom: 1 } })
})

test('handleGeneric', () => {
  const wrapper = shallow(<PasswordEntropyContainer />)
  wrapper.find('PasswordEntropy').simulate('generic', {
    target: { checked: false, name: 'Lowercase Letters' }
  })
  expect(wrapper.state()).toMatchObject({
    options: { Generic: { 'Lowercase Letters': false } }
  })
})

test('handleLength', () => {
  const wrapper = shallow(<PasswordEntropyContainer />)
  wrapper
    .find('PasswordEntropy')
    .simulate('length', { target: { value: '13' } })
  expect(wrapper.state('length')).toBe(13)
})

test('handleOptionsKey', () => {
  const wrapper = shallow(<PasswordEntropyContainer />)
  wrapper.find('PasswordEntropy').simulate('optionsKey', null, 'Diceware')
  expect(wrapper.state('optionsKey')).toBe('Diceware')
})
