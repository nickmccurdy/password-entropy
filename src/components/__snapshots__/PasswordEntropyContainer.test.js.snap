// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordEntropyContainer 1`] = `
<PasswordEntropy
  length={12}
  onCustom={[Function]}
  onGeneric={[Function]}
  onLength={[Function]}
  onOptionsKey={[Function]}
  options={
    Object {
      "Custom": 0,
      "Generic": Object {
        "Lowercase Letters": true,
        "Numbers": true,
        "Symbols": true,
        "Uppercase Letters": true,
      },
    }
  }
  optionsKey="Generic"
/>
`;
